keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/DependsOnConstraintEvaluator.kt

```diff
@@ -38,6 +40,27 @@ class DependsOnConstraintEvaluator(
       artifact,
       version,
       requiredEnvironment.name
+    ) && allVerificationsSucceededIn(
+      deliveryConfig,
+      artifact,
+      version,
+      requiredEnvironment
     )
   }
+
+  private fun allVerificationsSucceededIn(
+    deliveryConfig: DeliveryConfig,
+    artifact: DeliveryArtifact,
+    version: String,
+    environment: Environment
+  ): Boolean =
+    environment.verifyWith.all { verification ->
+      verificationRepository.wasSuccessfullyVerifiedIn(
+        deliveryConfig,
+        artifact.reference,
+        version,
+        environment.name,
+        verification
+      )
+    }
```

Comment:
This may be premature optimization but I wonder if we could query the status of all the verifications for the context and check that a) none are missing and b) they all passed. That way a single query could do the job.

Reply:
Done. I changed the verification repository method to `getStates`, which retrieves all of the states in associated with the context, and the DependsOnConstraintEvaluator does the check to see if any are missing.